### ISSUE TITLE
Genemodder trait : Atrophy - Breaks walk speed badly

### ISSUE THREAD

**Issue by SilvernKell** [BoHBranch/BoH-Bay#558]
#### Description of issue

A genemodder with the muscular atrophy trait appears to walk, run + sprint at a fraction of normal levels.
https://gyazo.com/0096b4ae3cee50d5ca5de0356b2251dd

#### Difference between expected and actual behavior

Instead of running, walking or sprinting at normal speeds -
They all default to the speed of crawl or probably even less.
Using Satchel, dropping bag makes no difference (does not seem to be linked to carry weight)

#### Steps to reproduce

Create a Genemodder
Use Muscular Atrophy trait
(Additional negative trait used at time of noticing bug was "incredibly frail")
(Positives were Agile, telepathy)

#### Specific information for locating

Negative trait - Muscular Atrophy
(Possibly interacts poorly with Agile trait?)

#### Length of time in which bug has been known to occur

This happened yesterday. (17th March)

#### Client version, Server revision & Game ID

Client Version: 512
Server Revision: 4d55a250f042d8227a74674dbdaee6fb09266460 -  - 2020-03-16

#### Issue bingo

- [x] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/Baystation12/Baystation12/issues)

**Comment by Roland410** [comment]
Are you sure you haven't used any other traits? Sluggish seems to be what you're having on that gif to be honest, not agile.

**Comment by BurgerLUA** [comment]
Has this bug occured WITHOUT the atrophy perk?

**Comment by bearrrrrrrr** [comment]
@BurgerLUA yeah, it occurs sometimes in general when someone comes out of cryo, a la the debuff never coming off them. i haven't been able to reproduce it.

**Comment by SilvernKell** [comment]
I got the bug and cryo'd and returned, same thing. I waited for another round to test it one more time and I had the same thing, so with muscular atrophy it was easy to replicate every time.
However every time I tested it I had agile on, so maybe the two are affecting each other

**Comment by Roland410** [comment]
Muscular Atrophy has nothing to do with walk speed, Agile does.

**Comment by SilvernKell** [comment]
Hence why it is a bug, my friend.

**Comment by MarinaGryphon** [comment]
I don't think this is a bug! The items you have on you may be too heavy for you. Does it still happen when completely naked?

**Comment by SilvernKell** [comment]
Yes it does

**Comment by MarinaGryphon** [comment]
So it turns out it applies the strength penalty to the slowdown, *permanently.* I guess you just don't have enough strength to lift your own body or something. Needs tweaking.

**Comment by Rafflesia7793** [comment]
This may be related.

Muscular Atrophy changes strength to STR_LOW

https://github.com/BoHBranch/BoH-Bay/blob/b95cdc2790810c43329ef9c3b62ab53dd4382cde/modular_mithra/code/modules/mithra/traits_negative.dm#L69-L73

STR_LOW is defined as -1.

https://github.com/BoHBranch/BoH-Bay/blob/6eda79470308696000e580c03f4ceded1e3d1bcb/code/__defines/mobs.dm#L131

Then slowdown is calculated as follows.

https://github.com/BoHBranch/BoH-Bay/blob/b95cdc2790810c43329ef9c3b62ab53dd4382cde/code/modules/mob/living/carbon/human/human_movement.dm#L46-L51

Is this literally a divide by zero error?
